test/e2e/windows/eviction.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package windows
+
+import (
+	"context"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/kubernetes/test/e2e/feature"
+	"k8s.io/kubernetes/test/e2e/framework"
+	e2enode "k8s.io/kubernetes/test/e2e/framework/node"
+	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+	imageutils "k8s.io/kubernetes/test/utils/image"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+const (
+	// It can take 10-15 seconds for node memory-pressure taint to show up on the node
+	// so we'll wait 20 seconds for the taint to show up so the e2e test case can catch
+	// it and the wait for the taint to be removed so other serial/slow tests can run
+	// against the same node.
```

Comment:
what happens if this test fails and doesn't remove the this taint? Can we add a deferred removal in the case where the tests fails, where it checks for the existence of and if still there removes it?

Reply:
The test doesn't remove the taint, the taint gets removed by the kubelet / eviction manager after the node is no longer experiencing memory / cpu pressure.
In this case I think we shouldn't manually remove the taint. If the test fails then the namespace created for the test will get deleted - which should delete the pods - and then everything should go back to normal after a few minutes.
The e2e test pass also will pause until all the nodes are schedulable again.

Let me know what you think

Reply:
This makes sense, thanks for explaining